ctype(3) functions used unsafely

<!--
READ THIS FIRST!

If you have a usage question, please ask us on [Stack Overflow](https://stackoverflow.com/questions/tagged/jq), in the [#jq channel](https://web.libera.chat/#jq) on [Libera.Chat](https://libera.chat/), or in our [Discord server](https://discord.gg/yg6yjNmgAC).

-->

**Describe the bug**
The ctype(3) functions are defined to only accept values in the range of `unsigned char` and `EOF` (despite having `int` in the prototype).
NetBSD is especially picky about this, but this can lead to real world problems e.g. with UTF-8 data.

**To Reproduce**
When compiling on NetBSD, the compiler warns about this:
```
In file included from /usr/include/ctype.h:100,
                 from src/main.c:2:
src/main.c: In function 'main':
src/main.c:440:23: warning: array subscript has type 'char' [-Wchar-subscripts]
  440 |               isspace(*argv[i+1]) || end == argv[i+1] || *end) {
      |                       ^
```

I think the use is safe here, so I suggest the following patch:
```
--- src/main.c.orig     2026-06-23 06:02:47.113070826 +0000
+++ src/main.c
@@ -437,7 +437,7 @@ int main(int argc, char* argv[]) {
           errno = 0;
           long indent = strtol(argv[i+1], &end, 10);
           if (errno || indent < -1 || indent > 7 ||
-              isspace(*argv[i+1]) || end == argv[i+1] || *end) {
+              isspace((unsigned char)*argv[i+1]) || end == argv[i+1] || *end) {
             fprintf(stderr, "jq: --indent takes a number between -1 and 7\n");
             die();
           }
```

**Expected behavior**
Safe use of ctype(3).

**Environment (please complete the following information):**

- OS and Version: NetBSD
- jq 1.8.2

**Additional context**
See the CAVEATS section of https://man.netbsd.org/ctype.3 for more details.

--- a/src/main.c
+++ b/src/main.c
@@ -437,7 +437,7 @@ int main(int argc, char* argv[]) {
           errno = 0;
           long indent = strtol(argv[i+1], &end, 10);
           if (errno || indent < -1 || indent > 7 ||
-              isspace(*argv[i+1]) || end == argv[i+1] || *end) {
+              isspace((unsigned char)*argv[i+1]) || end == argv[i+1] || *end) {
             fprintf(stderr, "jq: --indent takes a number between -1 and 7\n");
             die();
           }

